src/test/java/com/cloudant/tests/SearchTests.java

```diff
@@ -202,6 +201,25 @@ public void drillDownTest() {
         assertEquals(rslt.getRanges().get("min_length").get("large"), new Long(0));
     }
 
+    @Test
+    public void multipleDrillDownTest() {
+        // do a faceted search for drilldown
+        Search srch = db.search("views101/animals");
+        SearchResult<Animal> rslt = srch.includeDocs(true)
+                .counts(new String[]{"class", "diet"})
+                .ranges("{ \"min_length\": {\"small\": \"[0 TO 1.0]\","
+                        + "\"medium\": \"[1.1 TO 3.0]\", \"large\": \"[3.1 TO 9999999]\"} }")
+                .drillDown("class", "bird")
+                .drillDown("diet", "omnivore")
+                .querySearchResult("class:bird", Animal.class);
+        assertNotNull(rslt);
+        assertNotNull(rslt.getRanges());
+        assertEquals(rslt.getRanges().entrySet().size(), 1);
```

Comment:
If the JUnit assertions args need to be reversed in this test case, then we may need a work item to reverse all the other test cases.

And yes, the two drilldowns limits the returned result for the views101 design document to one single animal.